packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -0,0 +1,220 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import ssm = require('./ssm.generated');
+
+/**
+ * An SSM Parameter reference.
+ */
+export interface IParameter extends cdk.IConstruct {
+  /**
+   * The ARN of the SSM Parameter resource.
+   */
+  readonly parameterArn: string;
+
+  /**
+   * The name of the SSM Parameter resource.
+   */
+  readonly parameterName: string;
+
+  /**
+   * The type of the SSM Parameter resource.
+   */
+  readonly parameterType: string;
+
+  /**
+   * Grants read (DescribeParameter, GetParameter, GetParameterHistory) permissions on the SSM Parameter.
+   *
+   * @param grantee the role to be granted read-only access to the parameter.
+   */
+  grantRead(grantee: iam.IRole): IParameter;
+
+  /**
+   * Grants write (PutParameter) permissions on the SSM Parameter.
+   *
+   * @param grantee the role to be granted write access to the parameter.
+   */
+  grantWrite(grantee: iam.IRole): IParameter;
+}
+
+/**
+ * A String SSM Parameter.
+ */
+export interface IStringParameter extends IParameter {
+  /**
+   * A reference to the value of the SSM Parameter resource.
+   */
+  readonly parameterValue: string;
+}
+
+/**
+ * A StringList SSM Parameter.
+ */
+export interface IStringListParameter extends IParameter {
+  /**
+   * A reference to the value of the SSM Parameter resource.
+   */
+  readonly parameterValue: string[];
+}
+
+/**
+ * Properties needed to create a new SSM Parameter.
+ */
+export interface ParameterProps {
+  /**
+   * The allowed pattern for values of this parameter.
+   *
```

Comment:
Expand on the docs. Copy & paste from AWS Docs the details of how these patterns look